pkg/controller/cluster/cluster_controller.go

```diff
@@ -38,10 +38,9 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	if err != nil {
 		return err
 	}
-
 	return c.Watch(&source.Kind{Type: &synv1alpha1.GitRepo{}}, &handler.EnqueueRequestForOwner{
```

Comment:
Until we implemented #45 we could implement our own `handler.EnqueueRequestsFromMapFunc` to match on the `syn.tools/tenant` label.

This also applies to all the `EnqueueRequestForOwner` types of watches.

Reply:
I implemented #45 while debugging the reconcile lag for the tenant objects.

Reply:
Reading into the docs of `handler.EnqueueRequestsFromMapFunc` this could help solve the issues with the git repository lag I think.